src/webgl2_render_pass.rs

```diff
@@ -115,15 +142,21 @@ impl<'a> RenderPass for WebGL2RenderPass<'a> {
     }
 
     fn draw_indexed(&mut self, indices: Range<u32>, _base_vertex: i32, instances: Range<u32>) {
+        let (index_type, type_size) = match self.pipeline_descriptor.as_ref().unwrap().index_format
+        {
+            IndexFormat::Uint16 => (Gl::UNSIGNED_SHORT, 2),
+            IndexFormat::Uint32 => (Gl::UNSIGNED_INT, 4),
+        };
+
         let ctx = &self.render_context;
         let gl = &ctx.device.get_context();
         self.setup_vao();
         gl_call!(gl.draw_elements_instanced_with_i32(
             Gl::TRIANGLES,
-            indices.end as i32,
-            Gl::UNSIGNED_INT,
-            indices.start as i32,
-            instances.end as i32,
+            (indices.end - indices.start) as i32,
+            index_type,
+            indices.start as i32 * type_size,
+            (instances.end - instances.start) as i32,
         ));
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/drawElementsInstanced

**count:** we can't treat `indices.end as i32` as a count, otherwise users won't be able to use anything apart from ranges that start with `0`
**offset:** the documentation states that it "must be a valid multiple of the size of the given type"
**instanceCount:** same as `count`